### PR TITLE
Warn when ??? is used to implement erased definition

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala [hunk 1]
```diff
@@ -69,6 +73,12 @@ class PruneErasedDefs extends MiniPhase with SymTransformer { thisTransform =>
     else tree
 
   private def trivialErasedTree(tree: Tree)(using Context): Tree =
+    tree match
+      case tree: ValOrDefDef if !tree.symbol.is(Inline) && !tree.symbol.is(Synthetic) =>
+        if tree.rhs.symbol == defn.Predef_undefined then
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala:78]: Next step might be to do `tree.rhs.symbol != defn.Compiletime_erasedValue`.

**Review by liufengyun — COMMENTED** [review]
In #11721 , it uses `???` for erased definitions. Is there a potential conflict?

**Comment by nicolasstucki** [comment]
It is possible. Currently, it seems not non of the tests have fatal warning. It should be fine.

**Comment by liufengyun** [comment]
> It is possible. Currently, it seems not non of the tests have fatal warning. It should be fine.

In #11721 , it uses `erased given T = ???` to desugar `try/catch`.

**Comment by nicolasstucki** [comment]
That is not on the RHS of a definition. It will be fine.

**Comment by liufengyun** [comment]
> That is not on the RHS of an erased definition. It will be fine.

Could you elaborate? In `erased given T = ???`, it seems `???` is on the RHS.

**Pull request closed without merging**